unittests/runtime/CMakeLists.txt

```diff
@@ -1,13 +1,15 @@
 if(("${SWIFT_HOST_VARIANT_SDK}" STREQUAL "${SWIFT_PRIMARY_VARIANT_SDK}") AND
    ("${SWIFT_HOST_VARIANT_ARCH}" STREQUAL "${SWIFT_PRIMARY_VARIANT_ARCH}"))
 
-  if(SWIFT_BUILD_RUNTIME_WITH_HOST_COMPILER)
+  if("${CMAKE_C_COMPILER_ID}" MATCHES "Clang")
+    # Do nothing
+  elseif(SWIFT_BUILD_RUNTIME_WITH_HOST_COMPILER)
     if(NOT "${CMAKE_C_COMPILER_ID}" MATCHES "Clang")
       message(FATAL_ERROR "Building the swift runtime is not supported with ${CMAKE_C_COMPILER_ID}. Use the just-built clang instead.")
-    else()
-      message(WARNING "Building the swift runtime using the host compiler, and not the just-built clang.")
     endif()
   else()
+    message(WARNING "Building the swift runtime using the host compiler, and not the just-built clang.")
```

Comment:
Does this message actually belong here? Previously, the message was emitted when building with the host compiler and the host compiler is clang. This one will be emitted when not building with the host compiler, and the host compiler is not clang. But I may be misunderstanding, CMake gives me a headache.

Reply:
Yeah, that is correct, this is going to be emitted when building with the host compiler which is not clang, which is what we want to warn on.  I can tweak the message if you like, but, lets do that as a follow up, I would like to get the bots unblocked.

Reply:
If it's building with the host compiler, won't that take the above branch before the `else()`, where `elseif(SWIFT_BUILD_RUNTIME_WITH_HOST_COMPILER)`?

Totally fine with making adjustments in a followup. Or none at all if I'm just misunderstanding, which does happen sometimes.